### DISCUSSION TITLE
Intended way to retrieve a specific entity from an EGID?

### DISCUSSION THREAD

**Discussion by Danon5** [sebas77/Svelto.ECS#95] [Q&A] [answered]
Let's say there are 100 entities with the same descriptor (only has position component, for the sake of the hypothetical) in a networked game, and only one of them is moving. Because of delta compression, only the moving entity has its position sent. What would be the intended way to handle receiving and applying the position update to that particular entity?

 I would use QueryEntity, but the entity could come from one of a few groups, so I would have to iterate over each group. This means only one of the groups would have the entity. I can't null check because QueryEntity returns a struct. I would naturally look to TryGetEntity, but it does not seem to support unmanaged types, which means it cannot retrieve ViewComponents (which is what my desired component is).

**Comment by sebas77** [comment, marked as answer]
as long as you have an EGID available, you can query directly an Entity. There are two ways to do this:

first one is to use (Try)QueryEntitiesAndIndices

```
//this pattern KILLLS performance, components must be 1:1
            var gridRotationStructs   = entitiesDB.QueryEntitiesAndIndex<GridRotationStruct>(egid, out uint index);
            var (positions, _) = entitiesDB.QueryEntities<PositionEntityStruct>(groupID);
            var (rotations, _) = entitiesDB.QueryEntities<RotationEntityStruct>(groupID);
            var (scalings, _) = entitiesDB.QueryEntities<ScalingEntityStruct>(groupID);
            var (skews, _) = entitiesDB.QueryEntities<SkewComponent>(groupID);

            ref GridRotationStruct   gridRotationES = ref gridRotationStructs[index];
            ref PositionEntityStruct positionES     = ref positions[index];
            ref RotationEntityStruct rotationES     = ref rotations[index];
            ref ScalingEntityStruct  scalingES      = ref scalings[index];
            ref SkewComponent        skewComponent  = ref skews[index];
```

the reason to return the array and the indices is that then you can reuse the index for other arrays like in the example

second one is to use an EGIDMapper

```
 if (!entitiesDB.TryQueryMappedEntities(CommonExclusiveGroups.SIMULATION_BODIES_GROUP, out EGIDMapper<RigidBodyEntityStruct> rbMap))
                    return default;
ref var childRbComponent = ref rbMap.Entity( connectedBodiesBuffer[jointIndex].connectedEntityA);
```

**Reply by Danon5** [reply]
EGIDMapper is exactly what I needed, thanks.
